protocol/rep/rep.go

```diff
@@ -227,9 +225,8 @@ func (r *rep) RecvHook(m *mangos.Message) bool {
 	}
 	r.sock.SetSendError(nil)
 	r.backtraceL.Lock()
-	r.backtrace = append(r.backtracebuf[0:0], m.Header...)
+	r.backtrace = append(r.backtrace[0:0], m.Header...)
 	r.backtraceL.Unlock()
-	m.Header = nil
```

Comment:
Thank you for the response.

It is perhaps not so clear to see, since in this hunk I did two things, where I should probably only have done one:

1.  undo the `m.Header = nil` - I had checked the code, when the backtrace is later put back on again, there is no issue, since it uses the call `m.Header = append(m.Header[0:0], ...)`, i.e. it overwrites whatever the previous header was with the backtrace;
2. a small optimisation which combines `backtracebuf` and `bactrace` - thinking about the code, I thought why are they not one buffer, since `backtrace` is a reference to `backtracebuf` when this code happens:
```go
r.backtrace = append(r.backtracebuf[0:0], m.Header...)
```
According to the [append](https://golang.org/pkg/builtin/#append) documentation, the slice reference pointing to the original `backtrace` is, after the above line, replaced with a reference to `backtracebuf`, due to the return value of `append()`. That is the reason I combined both.